testing/implement-testing.md

```diff
@@ -0,0 +1,14 @@
+## Context
+Testing is an important part of application development cycle. Testing allows us to ensure that our software works according to the "specification".
+
+## Task
+As part of implementation of the REST api task, we will also be adding test for the application to ensure that our application behaves as intented. 
+We will do this by implementing unit and integration tests for the application.
+
+As part of the unit testing, we will write tests that validate the functionality of individual functions/methods.
+As part of integration testing, we will write tests the validate the functionality as a whole, for example making sure a REST endpoints returns the correct data or persists information correctly, when we "programmatically" send requests to the endpoint.
+
```

Comment:
should we add some commonly seen framework around for different languages? like mockito, junit, spock, cucumber for java, and something like wirmock as external help?

Reply:
Good point, I'll add some examples for java.